ndonnx/_core/_numericimpl.py

```diff
@@ -899,186 +745,124 @@ def shape(self, x):
         current = x
         while isinstance(current, ndx.Array):
             current = next(iter(current._fields.values()))
-        return _from_corearray(opx.shape(current))
-
-
-def _binary_op(
-    x: ndx.Array,
-    y: ndx.Array,
-    op: Callable[[_CoreArray, _CoreArray], _CoreArray],
-    via_dtype: dtypes.CoreType | None = None,
-):
-    return _variadic_op([x, y], op, via_dtype)
-
-
-def _unary_op(
-    x: ndx.Array,
-    op: Callable[[_CoreArray], _CoreArray],
-    via_dtype: dtypes.CoreType | None = None,
-) -> ndx.Array:
-    return _variadic_op([x], op, via_dtype)
-
-
-def _variadic_op(
-    args: list[Array],
-    op: Callable[..., _CoreArray],
-    via_dtype: dtypes.CoreType | None = None,
-    cast_return: bool = True,
-) -> ndx.Array:
-    args = promote(*args)
-    out_dtype = args[0].dtype
-    if not isinstance(out_dtype, (dtypes.CoreType, dtypes._NullableCore)):
-        raise TypeError(
-            f"Expected ndx.Array with CoreType or NullableCoreType, got {args[0].dtype}"
+        return from_corearray(opx.shape(current))
+
+    def can_cast(self, from_, to) -> bool:
+        if isinstance(from_, dtypes.CoreType) and isinstance(to, ndx.CoreType):
+            return np.can_cast(from_.to_numpy_dtype(), to.to_numpy_dtype())
+        return NotImplemented
```

Comment:
Just double checking: This is indeed what we want rather than returning `False` by default?

Reply:
I'd say so. It would be a little strange if you implement casting between a custom type and another type and then this returned `False`.